frontend/src/tests/application-tests.js

```diff
@@ -1,25 +1,268 @@
+            // create a user for the applicant to make sure they exist before changing the default user to applicant
+            const applicantAsUser = {
+                roles: ["ta"],
+            };
+            applicantAsUser.utorid = applicant.utorid;
+
+            resp = await apiPOST("/debug/users", applicantAsUser);
+            expect(resp).toHaveStatus("success");
+            Object.assign(applicantAsUser, resp.payload);
+
+            // set the default user to the applicant
+            resp = await apiPOST("/debug/active_user", applicantAsUser);
+            expect(resp).toHaveStatus("success");
+
+            // read survey.js posting data
+            resp = await apiGET(`/public/postings/${posting.url_token}`, true);
+            expect(resp).toHaveStatus("success");
+            checkPropTypes(surveyPropTypes, resp.payload);
+            surveyPrefill = resp.payload["prefilled_data"];
+
+            // restore the default user
+            resp = await apiPOST("/debug/active_user", defaultUser);
+            expect(resp).toHaveStatus("success");
+        });
+
+        // broken up into 2 tests: applicant & application(not yet implemented in the backend)
+        it("Survey.js posting data is pre-filled based on prior applicant", async () => {
+            // this test depends on the previous test on the terms of surveyPrefill variable
+            const prefilled = [
+                "utorid",
+                "student_number",
+                "first_name",
+                "last_name",
+                "email",
+                "phone",
+            ];
+            for (let prefilledKey of prefilled) {
+                // check if every applicant information is in the prefilled data
+                expect(surveyPrefill[prefilledKey]).toBeDefined();
+
+                // check if the applicant prefill information is correct
+                expect(surveyPrefill[prefilledKey]).toEqual(
+                    applicant[prefilledKey]
+                );
+            }
+        });
+
         it.todo(
-            "Survey.js posting data is pre-filled based on prior applicant/application data"
+            "Survey.js posting data is pre-filled based on prior application"
         );
+
         it.todo("Can submit survey.js data via the public postings route");
+
+        it("When submitting survey.js data an applicant and application are automatically created they don't exist", async () => {
+            // link seeded position to the posting (todo: remove after merged with previous test)
+            resp = await apiPOST(
+                `/admin/postings/${posting.id}/posting_positions`,
+                { ...postingPosition, position_id: position.id }
+            );
+            expect(resp).toHaveStatus("success");
+            Object.assign(postingPosition, resp.payload);
+
+            // create a new user
+            resp = await apiPOST("/debug/users", newUser);
```

Comment:
I'd rather have `newUser` defined in the test rather than at the beginning of the file.